### PR TITLE
55 add nix package manager

### PR DESCRIPTION
**Pull request by andriy-safe-ai**
I deployed in San Jose and ran the following tests:
Before I install Nix, I need to create a user to test whether Nix gets added to existing users. 
0. Install Nix
```
# run on bastion
# edit socat-server.service and socat-client.service files with the correct ip address
cd /opt/oci-hpc/playbook
ansible-playbook install_nix.yml
```
1. Make sure Nix is installed 
``` 
# run on login and backup
nix doctor
# expected output
[PASS] PATH contains only one nix version.
[PASS] All profiles are gcroots.
[PASS] Client protocol matches store protocol.
[INFO] You are trusted by store uri: daemon
…
```
2. Install a package
``` 
# run on login
nix profile install nixpkgs#hello
# test package
hello
>Hello, world!
```
3. Test package on different node
```
# run on bastion
hello
>Hello, world!
```
4. Repeat tests 1-3 as a user
5. Uninstall nix
```
# run on bastion
cd /opt/oci-hpc/playbooks
ansible-playbooks uninstall_nix.yml
```

### CODE REVIEW

File: playbooks/README.md
Comment (steven-basart) [inline playbooks/README.md:1]: replace "files" with file names to be more descriptive.

File: playbooks/files/socat-client.service
Comment (steven-basart) [inline playbooks/files/socat-client.service:1]: move the files directory into playbooks/roles/nix/files/ or playbooks/roles/nix/templates/

see [slurm folder](https://github.com/centerforaisafety/cerberus-cluster/tree/main/playbooks/roles/slurm) as an example

File: playbooks/roles/cais-all/tasks/ol-7.yml
Comment (steven-basart) [inline playbooks/roles/cais-all/tasks/ol-7.yml:1]: why is this line or file even edited?

File: playbooks/roles/nix/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        dest: /tmp
+        checksum: "{{ installer_checksum }}"
+    - name: extract installer
+      unarchive:
+        src: /tmp/{{ nix_build }}.tar.xz
+        remote_src: true
+        dest: /tmp/
+    - name: Run the installer
+      become: true
+      ansible.builtin.shell:
```
Comment (steven-basart) [inline playbooks/roles/nix/tasks/main.yml:17]: you can shorten this to just `shell`

**Review by steven-basart — CHANGES_REQUESTED** [review]
did a quick review.  Left a few comments.  

Also if you could please post your tests.

File: playbooks/files/socat-client.service
Comment (andriy-safe-ai) [reply playbooks/files/socat-client.service:1]: Even if the files aren't used in the role?

Make sense from since they're only used for Nix tho.

File: playbooks/roles/cais-all/tasks/ol-7.yml
Comment (andriy-safe-ai) [reply playbooks/roles/cais-all/tasks/ol-7.yml:1]: An idea that didn't work out. I'll remove any changes.

File: playbooks/roles/nix/tasks/main.yml (hunk 1 above)
Comment (andriy-safe-ai) [reply playbooks/roles/nix/tasks/main.yml:17]: 👍

File: playbooks/README.md
Comment (andriy-safe-ai) [reply playbooks/README.md:1]: sg

File: playbooks/files/socat-client.service
Comment (steven-basart) [reply playbooks/files/socat-client.service:1]: It's about keeping things organized. Having a general files folder doesn't help anyone when they need to search for the nix configurations and other related info.

File: playbooks/roles/nix/files/socat-client.service
Comment (steven-basart) [inline playbooks/roles/nix/files/socat-client.service:1]: You should be able to get the IP of the slurm backup node via Ansible. Add that as another issue. 

Change the IP to be ###.###.###.## so the files in nix role fails instead of trying to change a random IP.
